### PR TITLE
Move notebooks

### PR DESCRIPTION
**Pull request by gramster**
Move notebooks to separate git repo.

Disable save/rename when viewing a doc notebook to keep the repo clean.

Fix the 'Save a copy' menu item to copy samples to users datalab/notebooks folder.

### CODE REVIEW

File: sources/web/datalab/static/datalab.js [hunk 1]
```diff
@@ -173,15 +173,33 @@ function initializeNotebookApplication(ipy, notebook, events, dialog, utils) {
       return originalSave.apply(this, /* check_lastmodified */ [ false ]);
     }
 
+    function isSample() {
+      var path = window.location.pathname;
+      var lastSepPos = path.lastIndexOf('/');
+      return path.substring(lastSepPos-23, lastSepPos) == '/datalab/docs/notebooks';
```
Comment (qimingj) [inline sources/web/datalab/static/datalab.js]: Check lastSepPos > 23?

File: containers/datalab/content/run-debug.sh [hunk 2]
```diff
@@ -17,7 +17,13 @@
 
 mkdir -p /content/datalab/notebooks
 mkdir -p /content/datalab/docs
-rsync -r /datalab/docs/* /content/datalab/docs
+
+if [ -d /content/datalab/docs/.git ]
+then
+  (cd /content/datalab/docs; git fetch --all; git reset --hard origin/master)
```
Comment (qimingj) [inline containers/datalab/content/run-debug.sh]: Should it be "cd /content/datalab/docs/notebooks"? "Git clone" creates the notebooks folder and you have to be in that folder to run git fetch?

Comment (qimingj) [inline containers/datalab/content/run-debug.sh]: Consider fetch master only?

File: containers/datalab/content/run-debug.sh [hunk 3]
```diff
@@ -17,7 +17,13 @@
...
+else
+  (cd /content/datalab/docs; git clone https://github.com/googledatalab/notebooks.git)
```
Comment (qimingj) [inline containers/datalab/content/run-debug.sh]: Add "--branch master --single-branch" to get master branch only?

**Comment by qimingj** [comment]
A few comments

File: sources/web/datalab/static/datalab.js (hunk 1 above)
Comment (gramster) [inline sources/web/datalab/static/datalab.js]: I'll add that for clarity although it isn't actually necessary in Javascript. Anything less than zero is equivalent to 0 and the test still applies then.

File: containers/datalab/content/run-debug.sh (hunk 3 above)
Comment (gramster) [inline containers/datalab/content/run-debug.sh]: Done

File: containers/datalab/content/run-debug.sh (hunk 2 above)
Comment (gramster) [inline containers/datalab/content/run-debug.sh]: Done.

**Comment by qimingj** [comment]
LGTM!

**Comment by gramster** [comment]
PTAL

**Pull request merged**